lib/payrix/configuration.rb

```diff
@@ -3,20 +3,43 @@
 module Payrix
   # Use this class to configure API parameters such as API URL, API key, etc.
   class Configuration
-    attr_accessor :api_key, :session_key, :test_mode
+    attr_accessor :api_key, :session_key
+    attr_reader :environment
 
     def initialize
       @api_key = ''
       @session_key = ''
-      @test_mode = false
+      @environment = Payrix::ENVIRONMENTS.fetch(:sandbox)
     end
 
-    def url
-      if @test_mode
+    def environment=(environment)
+      validate_environment!(environment)
+
+      @environment = environment.to_sym
+    end
```

Comment:
Here I am defining an explicit setter so that we can raise an appropriate developer error when it is set incorrectly.